.github/workflows/publish-canaries.yml

```diff
@@ -0,0 +1,27 @@
+name: Publish canary packages to npm
+
+on:
+  push:
+    branches:
+      - master
+      - next
```

Comment:
1. doesn't work for us because of https://github.com/pnpm/pnpm/pull/7487#issuecomment-1889418444 and it doesn't seem like that [use-case has high priority](https://github.com/pnpm/pnpm/pull/7487#issuecomment-1933249584).
The problem is that we want a monorepo DX but not use a monorepo. It's a bit like wanting a monolith DX, while using microservices. Perhaps we should like into [bit](https://bit.dev/docs/thinking-bit/#bit-in-a-polyrepo)?

Reply:
Overall, the main point I wanted to make is that spamming the npm version page, would be nice to avoid but we can't sacrifice our ability to verify quickly propagate changes between repositories. If we have to sacrifice something, why not this. (in the past, it might have given the impression that no bloat in the npm package page was super important, so to be clear, no). 

> because of pnpm/pnpm#7487 (comment)

Right, but we don't necessarily have to use "workspace" in the dependency 😁. It's not as clear, but not a major cons. 

> For this I'd rather look into the direction of linking local dependencies. Building dependencies from source won't scale, it needs to be optional.

Right, but linking local dependencies only help with local development right? We still need to merge those changes in the main branch. First in Base UI, second in Material UI. Actually, sometimes, the fix in Material UI will be broken if it uses the wrong verison of Base UI, so we might even need to coordinate this deeper, wait Base UI release before updating Material UI. It's for this than monorepo are so much better, but yeah, we can't afford it for go to market reasons.

Reply:
> Right, but linking local dependencies only helps with local development?

yes

> We still need to merge those changes in the main branch. First in Base UI, and second in Material UI. Actually, sometimes, the fix in Material UI will be broken if it uses the wrong version of Base UI

* We develop the fix with local linking base.
* Once we verified the fix works both in base and in core, we can make the PR for base
* The PR in core can then use the codesandboxci published version of base, if you want to verify CI
* The fix in core can only be merged after the fix in base has been published.
* There is no other way in multiple repositories. otherwise you need to coordinate core and base publishing to the second.